src/cli/rustup_mode.rs

```diff
@@ -805,10 +805,17 @@ async fn check_updates(cfg: &Cfg<'_>, opts: CheckOpts) -> Result<utils::ExitCode
     // Ensure that `.buffered()` is never called with 0 as this will cause a hang.
     // See: https://github.com/rust-lang/futures-rs/pull/1194#discussion_r209501774
     if num_channels > 0 {
-        let multi_progress_bars = if is_a_tty {
-            MultiProgress::with_draw_target(ProgressDrawTarget::term_like(Box::new(t)))
-        } else {
-            MultiProgress::with_draw_target(ProgressDrawTarget::hidden())
+        let multi_progress_bars = match cfg.process.var("RUSTUP_TERM_PROGRESS_WHEN") {
+            Ok(s) if s.eq_ignore_ascii_case("always") => {
+                MultiProgress::with_draw_target(ProgressDrawTarget::term_like(Box::new(t)))
```

Comment:
Nit: @djc's previous comment applies: extract `MultiProgress::with_draw_target()` to outside of this `match`.